File: uLab/Assets/Scripts/UI/TipWindow.cs


using Locke;
using Locke.ui;
using UnityEngine;
using UnityEngine.UI;

public class TipWindow : IWindow
{
	protected override void OnInit()
	{
		Button button = this.FindWidget<Button>("buttons/button1");
		UIEventListener.Get(button.gameObject).onClick = onOkClick;

		Button back = FindWidget<Button>("buttons/back");
		back.onClick.AddListener(OnBackClick);
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

	private void onOkClick(GameObject go)
	{
		App.uiManager.CloseWindow(this);
	}

	private void OnBackClick()
	{
		App.uiManager.CloseWindow(this);
	}

}

File: uLab/Assets/Lua/UI/TipWindow.lua

local gameObject

local TipWindow = class("TipWindow",UIBase)

function TipWindow:OnInit(obj)
	gameObject = obj
	local ok = Util.FindGameObject(gameObject, 'buttons/button1')
	UIEventListener.SetOnClick(ok, onOkClick)
	local cancel = Util.FindGameObject(gameObject, 'buttons/back')
	UIEventListener.SetOnClick(cancel, onCancelClick)
end

function TipWindow:OnEnter()
	
end

function TipWindow:OnExit()
	
end

function TipWindow:OnPause()
	
end

function TipWindow:OnResume()
	
end

function onOkClick(go)
	CloseWindow(GameUI.tip)
end

function onCancelClick(go)
	CloseWindow(GameUI.tip)
end

return ShopWindow